### PR TITLE
Display Appointments and Patient Details Navigation with Camera Upload

### PR DESCRIPTION
**Pull request by pemba1s1**
**Issue**: #8 #10 #18 #21 

### Appointment Display on Main Page:
- Implemented functionality to display today's upcoming and completed appointments on the main page.
- Appointments are shown in a table format for easy viewing.

### Patient Details Navigation:
- Added the ability to click on an appointment in the table and navigate to the patient's details page using a new tab layout.
- The patient details page includes options to view various sections of the patient's profile.

### Camera Integration for Patient Demographics:
- From the patient's details page, added a camera tab that allows users to capture a photo and upload it directly to the patient's demographic information.
- Appointments Tab in Patient Details:
- Created a tab in the patient details layout to view all the upcoming and past appointments of the patient.

### Expected Behavior:
- Users can now view today's upcoming and completed appointments on the main page.
- Clicking on an appointment opens the corresponding patient’s details in a new tab layout.
- From the patient’s details page, users can:
- Navigate to the camera tab to take a picture and upload it to the patient’s demographic profile.
- Navigate to the appointment tab to see all past and upcoming appointments for that patient.

### ScreenShots
image
image
image
image

## Summary by Sourcery

Introduce features to display and navigate appointments and patient details, including camera integration for photo uploads. Refactor OAuthManager for improved API interaction and update project configuration to support new functionalities.

New Features:
- Implement appointment display on the main page, showing today's upcoming and completed appointments in a table format.
- Add navigation from the appointment table to a patient's details page using a new tab layout.
- Integrate a camera feature in the patient details page to capture and upload photos directly to the patient's demographic information.
- Create a tab in the patient details layout to view all upcoming and past appointments of the patient.

Enhancements:
- Refactor OAuthManager to use a new base URL format, replacing 'OSCAR' with 'O19'.
- Enhance the OAuthManager to return a custom response type for better error handling and success messages.

Build:
- Add 'expo-camera' plugin to the project configuration with camera permission settings.

Documentation:
- Update README to reflect the change from 'oscar' to 'O19' in the project description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # open-o-connect
 
-A mobile application for connecting to electronic medical record systems based on the open-source "oscar" project code.
+A mobile application for connecting to electronic medical record systems based on the open-source "O19" project code.
```
[the remark this answers is not in the document]
Comment (yingbull) [reply README.md:3]: I don't want to link to the oscar-emr.com website.

File: components/CameraComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
+import { useAuthManagerStore } from "@/store/useAuthManagerStore";
+import { StatusType } from "@/types/types";
+import { CameraView, CameraProps, useCameraPermissions } from "expo-camera";
+import { useState, useRef } from "react";
+import { Button, StyleSheet, Text, TouchableOpacity, View } from "react-native";
+
+export default function App({ demographicNo }: { demographicNo: number }) {
```
[the remark this answers is not in the document]
Comment (yingbull) [reply components/CameraComponent.tsx:7]: This seems reasonable. See what you can clean up a bit.

File: app/patient-detail/[id]/appointment.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  const [pastAppointments, setPastAppointments] = useState<Appointment[]>([]);
+  const [upcomingAppointments, setUpcomingAppointments] = useState<
+    Appointment[]
+  >([]);
+  const [loading, setLoading] = useState(true);
+  const { manager } = useAuthManagerStore();
+  const { id } = useLocalSearchParams();
+
+  useEffect(() => {
+    if (!manager) return;
```
[the remark this answers is not in the document]
Comment (yingbull) [reply app/patient-detail/[id]/appointment.tsx:36]: Yes, please do - just make sure it has the format right.

File: app/patient-detail/[id]/index.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+ * @returns {JSX.Element} The rendered component.
+ */
+const AppointmentDetail = () => {
+  const [patientDetail, setPatientDetail] = useState<PatientDetail>();
+  const [loading, setLoading] = useState(true);
+  const { manager } = useAuthManagerStore();
+  const { id } = useLocalSearchParams();
+
+  useEffect(() => {
+    if (!manager) return;
```
[the remark this answers is not in the document]
Comment (yingbull) [reply app/patient-detail/[id]/index.tsx:21]: yes please do.

File: app/patient-detail/[id]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        }
+      });
+  }, [id, manager]);
+
+  /**
+   * Renders the patient detail.
+   * @returns {JSX.Element|null} The rendered patient detail.
+   */
+  const ShowPatientDetail = () => {
+    if (!patientDetail) return null;
```
[the remark this answers is not in the document]
Comment (yingbull) [reply app/patient-detail/[id]/index.tsx:38]: Just make sure it is right.

File: components/CameraComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
+import { useAuthManagerStore } from "@/store/useAuthManagerStore";
```
Comment (yingbull) [inline components/CameraComponent.tsx:1]: This applies to all of it, really -
1) Unless you want something else, adopt Prettier for a code format standard for a project like this I think.   Plug-ins are easy and it should have good opinions.
2) Need more inline comments, and ideally, following a comment style that can be used to generate docs as well.  Pick the standard and go with it.

**Review by yingbull — CHANGES_REQUESTED** [review]
Look at the other Sourcery comments as well.

File: components/CameraComponent.tsx (hunk 6 above)
Comment (pemba1s1) [reply components/CameraComponent.tsx]: I added the jsdocs to all the files and prettier for code fomatting

**Review by yingbull — APPROVED** [review]

**Pull request merged**